bigquery_etl/cli/backfill.py

```diff
@@ -355,40 +350,46 @@ def process(ctx, qualified_table_name, sql_dir, project_id, dry_run):
     if backfills_to_process_dict:
         entry_to_process = backfills_to_process_dict[qualified_table_name]
 
-        backfill_staging_qualified_table_name = (
-            get_backfill_staging_qualified_table_name(
-                qualified_table_name, entry_to_process.entry_date
-            )
-        )
+        click.echo(f"\nProcessing backfills dry runs for {qualified_table_name}:")
+        _process_backfill(ctx, qualified_table_name, entry_to_process, dry_run=True)
 
-        project, dataset, table = qualified_table_name_matching(qualified_table_name)
+        click.echo(f"\nProcessing backfills for {qualified_table_name}:")
+        _process_backfill(ctx, qualified_table_name, entry_to_process)
 
-        click.echo(f"Processing backfills for {qualified_table_name}:")
+        click.echo(f"Backfill processing completed for {qualified_table_name}.")
 
-        # todo: send notification to watcher(s) that backill for file been initiated
 
-        ctx.invoke(
-            deploy,
-            name=f"{dataset}.{table}",
-            project_id=project,
-            destination_table=backfill_staging_qualified_table_name,
-        )
+def _process_backfill(ctx, qualified_table_name, entry_to_process, dry_run=None):
+    backfill_staging_qualified_table_name = get_backfill_staging_qualified_table_name(
+        qualified_table_name, entry_to_process.entry_date
+    )
 
-        # in the long-run we should remove the query backfill command and require a backfill entry for all backfills
-        ctx.invoke(
-            query_backfill,
-            name=f"{dataset}.{table}",
-            project_id=project,
-            start_date=entry_to_process.start_date,
-            end_date=entry_to_process.end_date,
-            exclude=entry_to_process.excluded_dates,
-            destination_table=backfill_staging_qualified_table_name,
-            dry_run=dry_run,
-        )
+    project, dataset, table = qualified_table_name_matching(qualified_table_name)
 
-        # todo: send notification to watcher(s) that backill for file has been completed
+    # deploy table
+    ctx.invoke(
+        deploy,
+        name=f"{dataset}.{table}",
+        project_id=project,
+        destination_table=backfill_staging_qualified_table_name,
+    )
```

Comment:
I vaguely remember discussing this but remind me, does dry-running the backfill query(ies) require the destination table to be deployed? Normally dry-running a query doesn't require the destination to exist.